Testscripts/Linux/perf_lkp.sh

```diff
@@ -0,0 +1,227 @@
+#!/bin/bash
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
+
+###############################################################################
+#
+# Description:
+# Support distros: Ubuntu 18.04 
+#
+###############################################################################
+
+LKP_SCHEDULER_HACKBENCH_TESTS=("hackbench-100.yaml" "hackbench.yaml")
+LKP_SCHEDULER_PERF_BENCH_SCHED_PIPE_TESTS=("perf-bench-sched-pipe.yaml")
+LKP_SCHEDULER_SCHBENCH_TESTS=("schbench.yaml")
+LKP_DATABASE_OLTP_TESTS=("oltp.yaml")
+LKP_DATABASE_REDIS_TESTS=("redis.yaml")
+LKP_MEMORY_PERF_BENCH_NUMA_MEM_TESTS=("perf-bench-numa-mem.yaml")
+LKP_MEMORY_PMBENCH_TESTS=("pmbench.yaml")
+LKP_WORKLOAD_APACHEBENCH_TESTS=("apachebench.yaml")
+LKP_TESTS_ALL_ABOVE=("perf-bench-sched-pipe.yaml" "schbench.yaml" "oltp.yaml" "redis.yaml" \
+                     "perf-bench-numa-mem.yaml" "pmbench.yaml" "apachebench.yaml")
+LKP_OTHER_UNIXBENCH_TESTS=("unixbench.yaml")
+LKP_OTHER_AIM9_TESTS=("aim9.yaml")
+
+###############################################################################
+function install_dependencies() {
+    LogMsg "Installing dependency packages"
+
+    GetDistro
+    update_repos
+
+    case $DISTRO in
+        ubuntu* | debian*)
+            CheckInstallLockUbuntu
+            deb_packages=(make git)
+            LogMsg "Dependency package names： ${deb_packages[@]}"
+            install_package "${deb_packages[@]}" >> $LKP_OUTPUT 2>&1
+            ;;
+        centos_7 | redhat_7)
+            rpm_packages=(make git curl psmisc)
+            install_epel
+            LogMsg "Dependency package names： ${rpm_packages[@]}"
+            install_package "${rpm_packages[@]}" >> $LKP_OUTPUT 2>&1
+            gpg --keyserver hkp://keys.gnupg.net --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3 7D2BAF1CF37B13E2069D6956105BD0E739499BDB
+            curl -sSL https://get.rvm.io | bash -s stable
+            if [ $? -ne 0 ]; then
+                LogErr "Failed to curl rvm"
+                SetTestStateAborted
+                exit 1
+            fi
+            source /etc/profile.d/rvm.sh
+            rvm requirements
+            rvm install ruby-2.3.0
+            rvm use ruby-2.3.0 default
+            PATH="$PATH:/usr/local/bin"
+            ;;
+        *)
+            LogErr "Unsupported distro: $DISTRO"
+            return 1
+            ;;
+    esac
+}
+
+function choose_test_case {
+    TEST_NAME_SET=("$TEST_NAME")
+    case $TEST_NAME in
+        hackbench)
+            LKP_TESTS=${LKP_SCHEDULER_HACKBENCH_TESTS[@]}
```

Comment:
use dict, no need to use so many switches?

Reply:
Thank you for this advice. I will use dict.